### ISSUE TITLE
Is there possibility to set a webhook for the fact of creating order (customer purchase) in my shop?

### ISSUE THREAD

**Issue by lofti198** [allegro/allegro-api#10512]
Hello! Is there possibility to set a webhook for the fact of creating order (customer purchase) in my shop? Please point to the documentation if available. Thank you

**Comment by lofti198** [comment]
How would you suggest to implement my task? Repeatedly check (poll) my orders? What method would recommend to use? Smth like this https://developer.allegro.pl/documentation#operation/getOrderEventsUsingGET ?

**Comment by MaciejFrackowiak** [comment]
Yes, you can monitor order changes using [GET /order/events](https://developer.allegro.pl/documentation/#operation/getOrderEventsUsingGET) resource. Please, check [our tutorial](https://developer.allegro.pl/tutorials/process-orders-PgPMlWDr8Cv#event-log) where we described how it works.

**Comment by MaciejFrackowiak** [comment]
You haven't switched the language version of the portal.
<img width="505" alt="Zrzut ekranu 2024-12-17 o 13 42 52" src="https://github.com/user-attachments/assets/e63ca5f8-c54f-4c7e-ad0b-e7eb6244c255" />

**Comment by lofti198** [comment]
Great thank you! One more thing - is there any option in API to get order events just for last 15 minutes for instance?

**Comment by lofti198** [comment]
What I still do not exaclty understand: can I get orders sorted from latest date? For instance if I setup limit up to 10 orders, they all be the latest ones? Or maybe just somehow get orders just from current date?

**Comment by MaciejFrackowiak** [comment]
You can use a filter "from" to receive only data from the last known event. Also, orders in the [GET /order/checkout-forms](https://developer.allegro.pl/documentation#operation/getListOfOrdersUsingGET) are returned from the newest one. You can use filters - "lineItems.boughtAt.gte" and "lineItems.boughtAt.lte" to specify the bought date.

**Comment by lofti198** [comment]
![image](https://github.com/user-attachments/assets/517a8a95-92a4-401c-81e2-cc9a69982ed7)
Ok I will try to save last ID and then pass it like this in request

**Comment by lofti198** [comment]
A question: for get order events https://developer.allegro.pl/documentation/#tag/Order-management limit is maximum 1000. And sorting is starting from earliest. What will happen if there are more than 1000 orders in my shop? Will the latest ones be absent in the response (and I will need to get them using already ?from parameter ?

**Comment by MaciejFrackowiak** [comment]
Order events return the last 1000 events, not orders - and yes, you have to use the parameter "from" to receive the latest one from a selected period.

**Comment by lofti198** [comment]
Hello! Why when we request status READY_FOR_PROCESSING we also get orders, which are not payed yet? https://developer.allegro.pl/documentation/#operation/getOrderEventsUsingGET

**Comment by MaciejFrackowiak** [comment]
Please, give me a trace-id of your request.

**Comment by lofti198** [comment]
Can I get full info about user just for particular order without iterating all checkout formst https://developer.allegro.pl/documentation#operation/getListOfOrdersUsingGET ? Just by order id for instance

![Image](https://github.com/user-attachments/assets/5a4f406f-b8a4-47b7-937d-12de16b871d3)

**Comment by MaciejFrackowiak** [comment]
Of course, you can retrieve a single order by [GET /order/checkout-forms/{id}](https://developer.allegro.pl/documentation/#operation/getOrdersDetailsUsingGET).
